### PR TITLE
Postgres: Views and named notations

### LINKED ISSUE #4069: (Postgres) Support CREATE OR REPLACE MATERIALIZED VIEW
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

`CREATE OR REPLACE VIEW` is already supported in Postgres, but `CREATE OR REPLACE MATERIALIZED VIEW` leads to:

```
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE OR
                       | REPLACE MATERIALIZED VIEW with...'
```

### Use case

_No response_

### Dialect

Postgres (with https://materialize.com/)

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### LINKED ISSUE #4072: A dot is valid in a parameter name for WITH in CREATE MATERIALIZED VIEW
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Follow [this tutorial](https://docs.timescale.com/timescaledb/latest/how-to-guides/continuous-aggregates/about-continuous-aggregates/) with:

```sql

CREATE MATERIALIZED VIEW daily_average WITH (timescaledb.continuous)
    AS SELECT time_bucket('1 day', time) as Day,
              location,
              avg(temperature)
       FROM temperatures
       GROUP BY day, location;
```

Then
```
sqlfluff  parse --dialect postgres timescale.sql 
echo $?
```

### Expected Behaviour

`0`

### Observed Behaviour

```
==== parsing violations ====
L:  98 | P:   1 |  PRS | Line 98, Position 1: Found unparsable section: 'CREATE
                       | MATERIALIZED VIEW daily_average\nWITH ...'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly
```

### How to reproduce

See above.

### Dialect

postgres

### Version

`1.4.2`

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### LINKED ISSUE #4074: Named notation => for Postgres functions
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Can't use [named notation](https://www.postgresql.org/docs/current/sql-syntax-calling-funcs.html#SQL-SYNTAX-CALLING-FUNCS-NAMED).

### Expected Behaviour

Parses

### Observed Behaviour

```
==== parsing violations ====
L: 142 | P:  31 |  PRS | Line 142, Position 31: Found unparsable section: " => 'Hello', b
                       | => 'World'"
```

### How to reproduce

Per [docs](https://www.postgresql.org/docs/current/sql-syntax-calling-funcs.html#SQL-SYNTAX-CALLING-FUNCS-NAMED):

```
SELECT concat_lower_or_upper(a => 'Hello', b => 'World');
 concat_lower_or_upper
-----------------------
 hello world
(1 row)
```

### Dialect

postgres

### Version

1.4.2

### Configuration

none

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by davetapley**
### Brief summary of the change made

Add support for conventions used by https://github.com/timescale/timescaledb:

Fixes #4072
Fixes #4074 

Since I was in there anyway:

Fixes #4069 

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - ~Full autofix test cases in `test/fixtures/linter/autofix`.~
  - ~Other.~
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by greg-finley** [comment]
Could you also run `black` formatting? https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes#black-code-linting

**Comment by jonatas** [comment]
Maybe it worth to mention that it supports the Timescaledb in the [dialects supported](https://github.com/sqlfluff/sqlfluff#dialects-supported) session?

**Comment by davetapley** [comment]
> Maybe it worth to mention that it supports the Timescaledb in the [dialects supported](https://github.com/sqlfluff/sqlfluff#dialects-supported) session?

@jonatas I'm not sure, since Timescaledb is really just a Postgres extension. Unless it has any of its own syntax?

**Comment by greg-finley** [comment]
Also, most people will just merge in `main` or push further commits based on feedback, instead of force-pushing. It's a bit easier to review that way.

**Comment by davetapley** [comment]
@greg-finley whoops, sorry about that. Linters run and rebased.

File: test/fixtures/dialects/postgres/postgres_create_materialized_view.sql [hunk 1]
```diff
@@ -145,3 +145,13 @@ FROM table_2
...
+
+CREATE MATERIALIZED VIEW my_mat_view
+WITH (left.right)
+AS
+SELECT a
+FROM my_table;
+
+CREATE OR REPLACE MATERIALIZED VIEW my_mat_view AS
+SELECT a
+FROM my_table;
```
Comment (greg-finley) [inline test/fixtures/dialects/postgres/postgres_create_materialized_view.sql:157]: Let's add newlines at end of files. Your IDE probably has a setting.

**Review by greg-finley — COMMENTED** [review]
Good start! Just a few minor things.

**Review by greg-finley — APPROVED** [review]
Thanks for the contribution and for working through the feedback! 🎉

**Comment by jonatas** [comment]
You're right @davetapley ! yeah, Timescaledb doesn't offers any dialect change.

I was thinking more on creating specific linters to Timescaledb functions, like when you create a refresh policy with interval smaller than the time_bucket, it would be great to just warn about it, but it would be necessary to have access to the time_bucket used in the view, so, probably worth to make it like a warning inside the function itself.

**Pull request merged**